pkg/profile/artifact.go

```diff
@@ -7,54 +7,180 @@ import (
+		if err := p.client.Create(ctx, helmRes); err != nil {
 			return fmt.Errorf("failed to create HelmRelease resource: %w", err)
 		}
-	case "Kustomize":
-		if err := p.createKustomization(ctx); err != nil {
+	} else {
+		p.log.Info("creating Kustomization", "resource", kustomizeRes.ObjectMeta.Name)
+		if err := p.client.Create(ctx, kustomizeRes); err != nil {
 			return fmt.Errorf("failed to create Kustomization resource: %w", err)
 		}
-	default:
-		return fmt.Errorf("artifact kind %q not recognized", kind)
 	}
-
 	p.log.Info("all artifacts created")
 	return nil
 }
 
+// Checks if the artifacts exists and returns any ready!=true conditions on the artifacts.
+func (p *Profile) ArtifactStatus(ctx context.Context) (Status, error) {
+	resourcesExist, gitRes, helmRes, kustomizeRes, err := p.getResources(ctx)
+	if err != nil {
+		return Status{}, err
+	}
+
+	return Status{
+		ResourcesExist:     resourcesExist,
+		NotReadyConditions: p.checkResourcesReady(gitRes, helmRes, kustomizeRes),
+	}, nil
+}
+
+func (p *Profile) checkResourcesReady(gitRes *sourcev1.GitRepository, helmRes *helmv2.HelmRelease, kustomizeRes *kustomizev1.Kustomization) []metav1.Condition {
+	var notReadyConditions []metav1.Condition
+	if gitRes != nil {
+		if condition := getNotReadyCondition(gitRes.Status.Conditions); condition != (metav1.Condition{}) {
+			notReadyConditions = append(notReadyConditions, condition)
+		}
+	}
+
+	if helmRes != nil {
+		if condition := getNotReadyCondition(helmRes.Status.Conditions); condition != (metav1.Condition{}) {
+			notReadyConditions = append(notReadyConditions, condition)
+		}
+	}
+
+	if kustomizeRes != nil {
+		if condition := getNotReadyCondition(kustomizeRes.Status.Conditions); condition != (metav1.Condition{}) {
+			notReadyConditions = append(notReadyConditions, condition)
+		}
+	}
+	return notReadyConditions
+}
+
+func (p *Profile) getResources(ctx context.Context) (bool, *sourcev1.GitRepository, *helmv2.HelmRelease, *kustomizev1.Kustomization, error) {
+	gitRes, helmRes, kustomizeRes, err := p.makeArtifacts()
+	if err != nil {
+		return false, nil, nil, nil, err
+	}
+	gitResExists := true
+	helmResExists := true
+	kustomizeResExists := true
+	if gitRes != nil {
+		gitResExists, err = p.getResourceIfExists(ctx, gitRes)
+		if err != nil {
+			return false, nil, nil, nil, err
+		}
+	}
+
+	if helmRes != nil {
+		helmResExists, err = p.getResourceIfExists(ctx, helmRes)
+		if err != nil {
+			return false, nil, nil, nil, err
+		}
+	}
+
+	if kustomizeRes != nil {
+		kustomizeResExists, err = p.getResourceIfExists(ctx, kustomizeRes)
+		if err != nil {
+			return false, nil, nil, nil, err
+		}
+	}
+	return gitResExists && helmResExists && kustomizeResExists, gitRes, helmRes, kustomizeRes, nil
+}
+
+func getNotReadyCondition(conditions []metav1.Condition) metav1.Condition {
+	for _, condition := range conditions {
+		if condition.Type == "Ready" && string(condition.Status) != "True" {
```

Comment:
https://pkg.go.dev/github.com/rmb938/hostport-allocator/api/meta#IsStatusConditionTrue ?

Reply:
oo nice! I'll refactor to use :smile: